sources/scraping/scrapy/pipeline_item.py

```diff
@@ -0,0 +1,19 @@
+from typing import Type
+
+import scrapy  # type: ignore
+from dlt.common import logger
+
+from .spider import DltSpider
+
+
+class ScrapingPipelineItem:
+    def process_item(
+        self, item: Type[scrapy.Item], spider: Type[DltSpider]
+    ) -> Type[scrapy.Item]:
+        """Handles sending the result of spider to scraping resource of Dlt pipeline"""
+        if not spider.queue.is_closed:
+            spider.queue.put(item)  # type: ignore
+        else:
+            logger.warning("Queue is closed")
```

Comment:
if a writer detects that queue is closed that means that reader died. so here we must stop crawling ie. by raising an exception or doing it some other recommended way (by scrapy)

Reply:
So now it will raise `CloseSpider` exception which should stop crawling.